frontends/web/src/app.tsx

```diff
@@ -168,6 +168,12 @@ class App extends Component<Props, State> {
         }
     }
 
+    // Returns a string representation of the current devices, so it can be used in the `key` property of subcomponents.
+    // The prefix is used so different subcomponents can have unique keys to not confuse the renderer.
+    private devicesKey = (prefix: string): string => {
+        return prefix + ':' + JSON.stringify(this.props.devices, Object.keys(this.props.devices).sort());
```

Comment:
Why not just `JSON.stringify(this.props.devices)`, why pass a sorted array of the keys as allowlist?

> A function that alters the behavior of the stringification process, or an array of String and Number that serve as an allowlist for selecting/filtering the properties of the value object to be included in the JSON string. If this value is null or not provided, all properties of the object are included in the resulting JSON string.

Reply:
I wasn't sure if the json serialization is deterministic. json object keys are not ordered in general, so in theory serialization could happen in any order.

Reply:
Ah I see, so but then just `JSON.stringify(Object.keys(this.props.devices).sort())`.  The second argument acts as a allowlist or does it have an effect on the order ?

Reply:
Good question. Your version does not include the values. Per documentation, the 2nd arg is only an allowlist, but in my tests at least it did serialize in that order (I took it from some stackoverflow post).

Reply:
yeah but the values are currently either 'bitbox01' or 'bitbox02' and the keys are already unique.